kubernetes/cache/cache.go

```diff
@@ -178,11 +192,27 @@ func (c *kialiCacheImpl) CheckNamespace(namespace string) bool {
 	return c.isKubernetesSynced(namespace) && c.isIstioSynced(namespace)
 }
 
+// RefreshNamespace will delete the specific namespace's cache and create a new one.
+func (c *kialiCacheImpl) RefreshNamespace(namespace string) {
+	if nsChan, exist := c.stopChan[namespace]; exist {
+		if nsChan != nil {
+			close(nsChan)
+			nsChan = nil //nolint
+		}
+	}
+	defer c.cacheLock.Unlock()
+	c.cacheLock.Lock()
+	delete(c.nsCache, namespace)
+	c.createCache(namespace)
+}
+
 func (c *kialiCacheImpl) Stop() {
 	log.Infof("Stopping Kiali Cache")
-	if c.stopChan != nil {
-		close(c.stopChan)
-		c.stopChan = nil
+	for _, nsChan := range c.stopChan {
```

Comment:
As similar as the other comment, I think that the stop channel should be protected by the lock.